### PR TITLE
Offers initial cleanups

### PR DESCRIPTION
**Pull request by rustyrussell**
~Based on #5605~ - Merged

This just does some preparatory cleanups in preparation for the spec-compliant implementation of blinded payments and modern offers.

### CODE REVIEW

**Comment by rustyrussell** [comment]
Trivial rebase on master now #5605 merged.

File: doc/lightning-makesecret.7.md [hunk 1]
```diff
@@ -35,7 +36,7 @@ Aditya <<aditya.sharma20111@gmail.com>> is mainly responsible.
 SEE ALSO
 --------
 
-lightning-getsharedsecret(7)
+lightning-datastore(7)
```
Comment (cdecker) [inline doc/lightning-makesecret.7.md:39]: Why is this referenced here?

**Review by cdecker — APPROVED** [review]
ACK 37456eb8587d8f3c78f753abfb18e111b7bbc34f

**Comment by rustyrussell** [comment]
Trivial rebase

Ack https://github.com/ElementsProject/lightning/commit/b0395224f7546ce62ae45dcbae8e3047dc9250a2

File: doc/lightning-makesecret.7.md (hunk 1 above)
Comment (rustyrussell) [reply doc/lightning-makesecret.7.md:39]: Hmm, mainly because it uses a similar API.  It *is* weird though, I'll remove it...

**Pull request merged**